zero_bin/ops/src/lib.rs

```diff
@@ -133,17 +133,20 @@ impl SegmentProofSpan {
         if ir.signed_txns.is_empty() {
             "Dummy".to_string()
         } else {
-            let first_encoding: [u8; 8] =
-                keccak(ir.signed_txns[0].clone())[..8].try_into().unwrap();
-            let last_encoding: [u8; 8] = keccak(
-                ir.signed_txns
-                    .last()
-                    .expect("the vector of transactions is not empty")
-                    .clone(),
-            )[..8]
-                .try_into()
-                .unwrap();
-            format!("{:x?}..{:x?}", first_encoding, last_encoding)
+            let first_encoding =
```

Comment:
The truncated hashes look good. But we still have the cumbersome `[`0xabcedf..0xabcedf]` for batches of length 1. Perhaps we could have 3 cases? (Dummy / single / multiple). Same goes with the `id()` which shows `0_1` / `1_2` / `2_3` while it may be clearer to only have single digits for single txns.